watcher/watcher.go

```diff
@@ -20,26 +20,82 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 )
 
+type WlidsToContainerToImageIDMap map[string]map[string]string
+
 const retryInterval = 3 * time.Second
 
 type WatchHandler struct {
 	k8sAPI                            *k8sinterface.KubernetesApi
 	imagesIDToWlidsMap                map[string][]string
-	wlidsToContainerToImageIDMap      map[string]map[string]string // <wlid> : <containerName> : imageID
-	imageIDsMapMutex                  *sync.Mutex
-	wlidsToContainerToImageIDMapMutex *sync.Mutex
+	instanceIDs                       []string
+	wlidsToContainerToImageIDMap      WlidsToContainerToImageIDMap // <wlid> : <containerName> : imageID
+	imageIDsMapMutex                  *sync.RWMutex
+	instanceIDsMutex                  *sync.RWMutex
+	wlidsToContainerToImageIDMapMutex *sync.RWMutex
 	currentPodListResourceVersion     string // current PodList version, used by watcher (https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes)
 }
 
+// remove unused imageIDs and instanceIDs from storage. Update internal maps
+func (wh *WatchHandler) cleanUp(ctx context.Context) {
+	storageImageIDs, err := wh.ListImageIDsFromStorage()
+	if err != nil {
+		logger.L().Ctx(ctx).Error("error to ListImageIDsFromStorage", helpers.Error(err))
```

Comment:
does it make sense to continue after this error?

Reply:
IMO it makes sense, since on the continuation we will list all Pods and according to that build the internal map. That way workloads that have been deleted will be erased from the map. 
If I get an error there (listing Pods), then I won't continue

Reply:
pls add a comment with justification